source/as-promise.ts

```diff
@@ -40,6 +40,11 @@ export default function asPromise(options: Options) {
 				return;
 			}
 
+			if (response.canceled()) {
```

Comment:
If the cached requests are not cancelable we could simply do `if (response.req && response.req.aborted) {`